### PR TITLE
[Merged by Bors] - feat: `isCoprime_mul_units_left`, `isCoprime_mul_units_right`

### PR DESCRIPTION
**Pull request by seewoo5**
---

Add `isCoprime_mul_units_left` and `isCoprime_mul_units_right`. These are slight generalizations of same unit version `isCoprime_mul_unit_left` and `isCoprime_mul_unit_right`. Currently exists as a lemma in #18882.

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by seewoo5** [comment]
There are two things that I can do further (but not sure if it worths to do):

- I can replace the existing proofs of `isCoprime_mul_unit_left` and `isCoprime_mul_unit_right` with `units` version
- There are two other possible combinations: `u * y, z * v` and `y * u, v * z`. I'm not sure if it is necessary to add these.

File: Mathlib/RingTheory/Coprime/Basic.lean [hunk 1]
```diff
@@ -244,7 +244,7 @@ end ScalarTower
 
 section CommSemiringUnit
 
-variable {R : Type*} [CommSemiring R] {x : R}
+variable {R : Type*} [CommSemiring R] {x u v: R}
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Coprime/Basic.lean:247]:
```suggestion
variable {R : Type*} [CommSemiring R] {x u v : R}
```

File: Mathlib/RingTheory/Coprime/Basic.lean [hunk 2]
```diff
@@ -272,6 +272,18 @@ theorem isCoprime_mul_unit_right (hu : IsUnit x) (y z : R) :
     IsCoprime (y * x) (z * x) ↔ IsCoprime y z :=
   (isCoprime_mul_unit_right_left hu y (z * x)).trans (isCoprime_mul_unit_right_right hu y z)
 
+theorem isCoprime_mul_units_left (hu : IsUnit u) (hv: IsUnit v) (y z: R) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Coprime/Basic.lean:275]:
```suggestion
theorem isCoprime_mul_units_left (hu : IsUnit u) (hv : IsUnit v) (y z : R) :
```

File: Mathlib/RingTheory/Coprime/Basic.lean [hunk 3]
```diff
@@ -272,6 +272,18 @@ theorem isCoprime_mul_unit_right (hu : IsUnit x) (y z : R) :
...
+    IsCoprime (u * y) (v * z) ↔ IsCoprime y z :=
+  Iff.trans
+    (isCoprime_mul_unit_left_left hu _ _)
+    (isCoprime_mul_unit_left_right hv _ _)
+
+theorem isCoprime_mul_units_right (hu : IsUnit u) (hv: IsUnit v) (y z : R) :
```
Comment (Ruben-VandeVelde) [inline Mathlib/RingTheory/Coprime/Basic.lean:281]:
```suggestion
theorem isCoprime_mul_units_right (hu : IsUnit u) (hv : IsUnit v) (y z : R) :
```

**Review by Ruben-VandeVelde — COMMENTED** [review]
I probably would suggest reproving `isCoprime_mul_unit_left/right`, yes

File: Mathlib/RingTheory/Coprime/Basic.lean [hunk 4]
```diff
@@ -268,9 +264,25 @@ theorem isCoprime_mul_unit_right_right (hu : IsUnit x) (y z : R) :
...
+    (isCoprime_mul_unit_right_right hv _ _)
+
+theorem isCoprime_mul_unit_left (hu : IsUnit x) (y z : R) :
+    IsCoprime (x * y) (x * z) ↔ IsCoprime y z :=
+  isCoprime_mul_units_left hu hu _ _
+
 theorem isCoprime_mul_unit_right (hu : IsUnit x) (y z : R) :
     IsCoprime (y * x) (z * x) ↔ IsCoprime y z :=
-  (isCoprime_mul_unit_right_left hu y (z * x)).trans (isCoprime_mul_unit_right_right hu y z)
+  isCoprime_mul_units_right hu hu _ _
```
Comment (tb65536) [inline Mathlib/RingTheory/Coprime/Basic.lean:285]: Would it be better to remove these two lemmas and add deprecated aliases pointing to your lemmas?

Comment (Ruben-VandeVelde) [reply Mathlib/RingTheory/Coprime/Basic.lean:285]: No, I don't think so

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks! 

maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**